Modules/File/classes/class.ilObjFile.php

```diff
@@ -678,16 +675,34 @@ function sendFile($a_hist_entry_id = null)
 		if (@is_file($file))
 		{
 			global $ilClientIniFile;
-			
-			// also returning the 'real' filename if a history file is delivered
-			if ($ilClientIniFile->readVariable('file_access','download_with_uploaded_filename') != '1' && is_null($a_hist_entry_id))
-			{
-				ilUtil::deliverFile($file, $this->getTitle(), $this->guessFileType($file), $this->isInline());
+			/**
+			 * @var $ilClientIniFile ilIniFile
+			 */
+			require_once('./Services/FileDelivery/classes/class.ilFileDelivery.php');
+
+			$ilFileDelivery = new ilFileDelivery($file);
```

Comment:
We may also consider to introduce a factory, before even more places make use of this.